File: Temp/leetcode/editor/cn/[LCR 036]逆波兰表达式求值.c
// LCR 036 逆波兰表达式求值
// https://leetcode.cn/problems/8Zf90G/

//leetcode submit region begin(Prohibit modification and deletion)
#include <string.h>
#include <stdlib.h>
bool isNum(char* token){
    return strlen(token) > 1 ||  token[0] <= '9' &&  token[0] >= '0';
}
int evalRPN(char** tokens, int tokensSize){
    int st[tokensSize], top = 0;
    for(int i = 0; i < tokensSize; i++){
        char* token = tokens[i];
        if (isNum(token)){
            st[top++] = atoi(token);
        }else{
            int num2 = st[--top];
            int num1 = st[--top];
            switch (token[0]){
                case '+':
                    st[top++] = num1 + num2;
                    break;
                case '-':
                    st[top++] = num1 - num2;
                    break;
                case '*':
                    st[top++] = num1 * num2;
                    break;
                case '/':
                    st[top++] = num1 / num2;
                    break;
            }
        }
    }
    return st[0];
}
//leetcode submit region end(Prohibit modification and deletion)

File: Temp/leetcode/editor/cn/[LCR 036]逆波兰表达式求值.py
# LCR 036 逆波兰表达式求值
# https://leetcode.cn/problems/8Zf90G/
from imports import *


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for s in tokens:
            if s not in "+-*/":
                st.append(int(s))
                continue
            a = st.pop()
            b = st.pop()
            match s:
                case "+":
                    st.append(a + b)
                case "-":
                    st.append(b - a)
                case "*":
                    st.append(a * b)
                case "/":
                    st.append(int(b / a))
        return st[-1]
    # leetcode submit region end(Prohibit modification and deletion)
